contracts/rosca.sol

```diff
@@ -1,72 +1,266 @@
-    
-    event LogParticipantRegistered(string uniqName);
-    event LogContributionMade(string uniqName);
-    event LogBidPlaced(string uniqName);
-    event LogRoundFundsRelease(string winnerUniqName, uint amountInWei);
-    
-    
-    enum Period { Weekly, Monthly }
-    
+    address constant WETRUST = 0x0;
+
+    enum Period { Weekly, Monthly } // are we only allwing 2 options?
+
+    event LogParticipantRegistered(address user);
+    event LogContributionMade(address user, uint amount);
+    event LogNewLowestBid(uint bid,address winnerAddress);
+    event LogRoundFundsRelease(address winnerUniqName, uint amountInWei);
+
+
+    //added
+    event LogFundsWithdrawal(address user, uint amount,address destination);
+    event LogStartOfPeriod(uint currentRound);
+
+    //state variables
+    address foreman;
+    uint roundSum;
+    uint numRounds;
+    uint currentEpoch;
+    uint numEpochs;
+    int8 minParticipants;
+    uint roundPeriodInDays;
+    uint startTime;
+    int16 feeToService;
+    address feeAddress;
+    uint currentRound;  //currentRound will be set to 0 when ROSCA is created and will turn to one when the ROSCA actually start
+
+    /*
+        not sure about which structures to use yet
+    */
+    struct USER{
+        uint contributed; // number of times contributed or could be amount contributed
+        uint paid; // number of times bid Won
+        uint allowance; // how much they are allowed to withdraw, i.e if someone win , their allowance will go up by the bid.
+        bool alive; //needed to check if a member is indeed a member
+
+    }
+    mapping(address => USER) members; //using struct USER to keep track of contributions and paid, allowance and etc.
+    address[] memberAddress;    //this is the only way to iterate through all the member's address
+
+    mapping(address => bool) pendingJoinRequest; // this way , address can be used as index, if we use address[] , we'll have to go thru a whole array
+
+    //deposit and withdrawal related
+    uint totalAmountInContract;
+    //bidding related state variable
+    uint lowestBid;
+    address winnerAddress;
+
+
     modifier onlyForeman {
+        if(msg.sender != foreman)
+        throw;
+        _;
+    }
+    modifier beforeStart {
+        if(currentRound == 0)
+        throw;
         _;
     }
-    
+
     /**
      * Creates a new ROSCA in which every roundPeriod each person contributes
      * roundSum, for numRounds
      */
-    function ROSCA(
-        Period roundPeriod, 
-        uint roundSum, 
-        int16 numRounds, 
-        int8 minParticipants, 
-        int startTime, 
-        int16 feeInThousandths, 
-        address feeAddress);
-    
+    function rosca(
+        uint _roundPeriodInDays,
+        uint _roundSum, //i am assuming this is referring to number epochs to run
+        uint _numEpochs,
+        int8 _minParticipants,
+        uint _startTime,
+        int16 _feeInThousandths,
+        address _feeAddress)
```

Comment:
I don't remember exactly what we discussed back in Starbucks - shouldn't this be a constant?